### PR TITLE
Load tools with collection type source attributes into the workflow editor

### PR DESCRIPTION
**Pull request by jmchilton**
Should allow using filter failed and merge collection tools within workflows.

This is really a very rough first order approximation of how the editor should be reasoning about the collection types and mapping for these tools - the way it reasons about format_source for datatypes for non-collection connections - but it is probably better than just excluding such connections?

Fixes https://github.com/galaxyproject/galaxy/issues/4509 ... at least partially.

### CODE REVIEW

**Comment by mvdbeek** [comment]
This works nicely, and it actually also makes other (in my instance all!) collection operation tools available in the workflow editor.

File: client/galaxy/scripts/mvc/workflow/workflow-terminals.js [hunk 1]
```diff
@@ -476,11 +476,29 @@ define(['mvc/workflow/workflow-globals'], function( Globals ) {
         initialize: function( attr ) {
             Terminal.prototype.initialize.call( this, attr );
             this.datatypes = attr.datatypes;
-            this.collectionType = new CollectionTypeDescription( attr.collection_type );
-            this.isCollection = true;
+            if( attr.collection_type ) {
+                this.collectionType = new CollectionTypeDescription( attr.collection_type );
+            } else {
+                var collectionTypeSource = attr.collection_type_source;
+                if(! collectionTypeSource) {
```
Comment (mvdbeek) [inline client/galaxy/scripts/mvc/workflow/workflow-terminals.js]: Here and in the update function there is a space missing before the closing parens.

**Review by mvdbeek — APPROVED** [review]
Looks good to me.

**Comment by jmchilton** [comment]
I adjusted the style a bit - but I'm not sure we want that extra space. I get that it is inconsistent right now - hopefully we will be applying a fixed style soon.

Otherwise - thanks for the detailed review @mvdbeek - I really appreciate it and I'm glad it worked.

**Comment by mvdbeek** [comment]
Thanks a lot @jmchilton !

**Pull request merged**